src/robust.jl

```diff
@@ -0,0 +1,98 @@
+# Robust Statistics
+
+#############################
+#
+#   Trimming outliers
+#
+#############################
+
+# Trimmed set
+"""
+    trim(x, p=0.2)
+
+Remove a proportion `p` of its highest elements, and `p` of its lowest elements and return the result. If you want to compute the variance of `mean(trim(x,p))`, use `trimvar`.
+
+# Example
+```julia
+julia> trim([1,2,3,4,5] , 0.2)
+ 2
+ 3
+ 4
+```
+"""
+trim(x,p::Real=0.2) = trim!(copy(x),p)
+trim!(x,p::Real=0.2) = trim!(collect(x), p)
+
+"""
+    trim!(x, p=0.2)
+
+A variant of `trim` that modifies vector `x` in place. See also `trimvar`.
+"""
+function trim!(x::RealArray, p::Real=0.2)
+    n = length(x)
+    n > 0 || error("x can not be empty.")
+    0 <= p < 0.5 || error("p must be in 0 <= p < 0.5.")
+    g = floor(Int, n * p)
+    
+    return select!(x, (g+1):(n-g))
```

Comment:
I think it would be faster to use the same strategy as in `winsor` by calling `select!` twice, for lowest and highest elements, and then indexing into the result.